imports/ui/components/root-app-bar.jsx

```diff
@@ -1,36 +1,65 @@
-import React from 'react'
+import React, { Component } from 'react'
 import PropTypes from 'prop-types'
 
 import AppBar from 'material-ui/AppBar'
 import IconButton from 'material-ui/IconButton'
 import FontIcon from 'material-ui/FontIcon'
 import { UneeTIcon } from '../components/unee-t-icons'
+import SearchBar from './search-bar'
 
 import {
   titleStyle,
   logoIconStyle,
   logoButtonStyle
 } from '../components/app-bar.mui-styles'
 
-const RootAppBar = ({ title, onIconClick, shadowless }) => (
-  <AppBar title={title} titleStyle={titleStyle} style={shadowless ? {boxShadow: 'none'} : undefined}
-    iconElementLeft={
-      <IconButton iconStyle={logoIconStyle} style={logoButtonStyle} onClick={onIconClick}>
-        <UneeTIcon />
-      </IconButton>
+class RootAppBar extends Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      searchTextDisplay: false
     }
-    iconElementRight={
-      <div>
-        <IconButton>
-          <FontIcon className='material-icons' color='white'>search</FontIcon>
-        </IconButton>
-        <IconButton>
-          <FontIcon className='material-icons' color='white'>notifications</FontIcon>
-        </IconButton>
-      </div>
-    }
-  />
-)
+  }
+  render () {
+    const { title, onIconClick, shadowless, findUnit, searchOff } = this.props
+    const { searchTextDisplay } = this.state
+
+    return (
+      <AppBar
+        title={searchTextDisplay ? <SearchBar findUnit={findUnit} /> : title}
+        id={title}
+        titleStyle={titleStyle}
+        style={shadowless ? {boxShadow: 'none'} : undefined}
+        iconElementLeft={
+          searchTextDisplay ? (
+            <IconButton
+              onClick={() => { this.setState({searchTextDisplay: false}); searchOff() }}
+            >
+              <FontIcon className='material-icons' color='white'>
+               arrow_back</FontIcon>
+            </IconButton>
+          ) : (
+            <IconButton iconStyle={logoIconStyle} style={logoButtonStyle} onClick={onIconClick}>
+              <UneeTIcon />
+            </IconButton>
+          )
+        }
+        iconElementRight={
+          <div>
+            <IconButton onClick={() => title === 'My Units' && this.setState({searchTextDisplay: true})}>
```

Comment:
Change this check of whether the `title` equals a specific value to checking a configurable boolean prop